dotnet migrate: buildOptions/languageVersion is not migrated properly

## Steps to reproduce

1. `dotnet new`
1. Add `"languageVersion": "csharp5"` to the `"buildOptions"` node.
1. `dotnet migrate`
1. `dotnet restore3`
1. `dotnet build3`

## Expected  behavior

The build should succeed. The .csproj should contain `<LangVersion>5</LangVersion>`.

## Actual behavior

The build fails:

```
CSC : error CS1617: Invalid option 'csharp5' for /langversion; must be ISO-1, ISO-2, Default or an integer in range 1 to 6. [C:\Users\joelv\Desktop\migrate\langver\langver.csproj]
```

## Workaround

Edit the .csproj `<LangVersion>csharp5</LangVersion>` to `<LangVersion>5</LangVersion>`.

## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (1.0.0-preview3-003830)

Product Information:
 Version:            1.0.0-preview3-003830
 Commit SHA-1 hash:  90a51bdd00

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
```

--- a/src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs
+++ b/src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs
@@ -76,7 +76,7 @@ namespace Microsoft.DotNet.ProjectJsonMigration.Rules
 
         private AddPropertyTransform<CommonCompilerOptions> LanguageVersionTransform =>
             new AddPropertyTransform<CommonCompilerOptions>("LangVersion",
-                compilerOptions => compilerOptions.LanguageVersion,
+                compilerOptions => FormatLanguageVersion(compilerOptions.LanguageVersion),
                 compilerOptions => !string.IsNullOrEmpty(compilerOptions.LanguageVersion));
 
         private AddPropertyTransform<CommonCompilerOptions> DelaySignTransform =>
@@ -354,6 +354,16 @@ namespace Microsoft.DotNet.ProjectJsonMigration.Rules
                     new JObject(),
                     null,
                     ProjectFilesCollection.DefaultPublishExcludePatterns);
+        }
+
+        private string FormatLanguageVersion(string langVersion)
+        {
+            if (langVersion.StartsWith("csharp"))
+            {
+                return langVersion.Replace("csharp", "");
+            }
+
+            return langVersion;
         }
     }
 }

--- a/src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs
+++ b/src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateBuildOptionsRule.cs
@@ -358,9 +358,9 @@ namespace Microsoft.DotNet.ProjectJsonMigration.Rules
 
         private string FormatLanguageVersion(string langVersion)
         {
-            if (langVersion.StartsWith("csharp"))
+            if (langVersion.StartsWith("csharp", StringComparison.OrdinalIgnoreCase))
             {
-                return langVersion.Replace("csharp", "");
+                return langVersion.Substring("csharp".Length);
             }
 
             return langVersion;

--- a/src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateConfigurationsRule.cs
+++ b/src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateConfigurationsRule.cs
@@ -56,7 +56,7 @@ namespace Microsoft.DotNet.ProjectJsonMigration.Rules
             MigrationRuleInputs migrationRuleInputs)
         {
             var buildOptions = project.GetRawCompilerOptions(framework);
-            var configurationCondition = $" '$(TargetFrameworkIdentifier),Version=$(TargetFrameworkVersion)' == '{framework.DotNetFrameworkName}' ";
+            var configurationCondition = framework.GetMSBuildCondition();
 
             MigrateConfiguration(buildOptions, configurationCondition, migrationSettings, migrationRuleInputs);
         }

--- a/src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateProjectDependenciesRule.cs
+++ b/src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateProjectDependenciesRule.cs
@@ -103,7 +103,7 @@ namespace Microsoft.DotNet.ProjectJsonMigration.Rules
         {
             if (targetFramework != null)
             {
-                itemGroup.Condition = $" '$(TargetFrameworkIdentifier),Version=$(TargetFrameworkVersion)' == '{targetFramework.DotNetFrameworkName}' ";
+                itemGroup.Condition = targetFramework.GetMSBuildCondition();
             }
 
             foreach (var projectDependencyTransformResult in projectDependencyTransformResults)

--- a/src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateProjectDependenciesRule.cs
+++ b/src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateProjectDependenciesRule.cs
@@ -32,8 +32,6 @@ namespace Microsoft.DotNet.ProjectJsonMigration.Rules
             var migratedXProjDependencyPaths = MigrateXProjProjectDependencies(migrationRuleInputs);
             var migratedXProjDependencyNames = new HashSet<string>(migratedXProjDependencyPaths.Select(p => Path.GetFileNameWithoutExtension(
                                                                                                                  PathUtility.GetPathWithDirectorySeparator(p))));
-
-            AddPropertyTransformsToCommonPropertyGroup(migrationRuleInputs.CommonPropertyGroup);
             MigrateProjectJsonProjectDependencies(
                 migrationRuleInputs.ProjectContexts, 
                 migratedXProjDependencyNames, 
@@ -113,30 +111,6 @@ namespace Microsoft.DotNet.ProjectJsonMigration.Rules
                 _transformApplicator.Execute(projectDependencyTransformResult, itemGroup);
             }
         }
-
-        private void AddPropertyTransformsToCommonPropertyGroup(ProjectPropertyGroupElement commonPropertyGroup)
-        {
-            var propertyTransformResults = new[]
-            {
-                AutoUnifyTransform.Transform(true),
-                DesignTimeAutoUnifyTransform.Transform(true)
-            };
-
-            foreach (var propertyTransformResult in propertyTransformResults)
-            {
-                _transformApplicator.Execute(propertyTransformResult, commonPropertyGroup);
-            }
-        }
-
-        private AddPropertyTransform<bool> AutoUnifyTransform => new AddPropertyTransform<bool>(
-            "AutoUnify",
-            "true",
-            b => true);
-
-        private AddPropertyTransform<bool> DesignTimeAutoUnifyTransform => new AddPropertyTransform<bool>(
-            "DesignTimeAutoUnify",
-            "true",
-            b => true);
 
         private AddItemTransform<ProjectDependency> ProjectDependencyTransform => new AddItemTransform<ProjectDependency>(
             "ProjectReference",

--- a/test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigrateBuildOptions.cs
+++ b/test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigrateBuildOptions.cs
@@ -240,6 +240,20 @@ namespace Microsoft.DotNet.ProjectJsonMigration.Tests
         }
 
         [Fact]
+        public void Migrating_languageVersion_removes_csharp_in_LangVersion()
+        {
+            var mockProj = RunBuildOptionsRuleOnPj(@"
+                {
+                    ""buildOptions"": {
+                        ""languageVersion"": ""csharp5""
+                    }
+                }");
+
+            mockProj.Properties.Count(p => p.Name == "LangVersion").Should().Be(1);
+            mockProj.Properties.First(p => p.Name == "LangVersion").Value.Should().Be("5");
+        }
+
+        [Fact]
         public void Migrating_keyFile_populates_AssemblyOriginatorKeyFile_and_SignAssembly()
         {
             var mockProj = RunBuildOptionsRuleOnPj(@"

--- a/test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigrateConfigurations.cs
+++ b/test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigrateConfigurations.cs
@@ -59,7 +59,7 @@ namespace Microsoft.DotNet.ProjectJsonMigration.Tests
 
             mockProj.Properties.First(p => p.Name == "OutputType")
                 .Parent.Condition.Should()
-                .Contain("'$(TargetFrameworkIdentifier),Version=$(TargetFrameworkVersion)' == '.NETCoreApp,Version=v1.0'");
+                .Contain("'$(TargetFramework)' == 'netcoreapp1.0'");
         }
 
         [Fact]

--- a/test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigrateProjectDependencies.cs
+++ b/test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigrateProjectDependencies.cs
@@ -16,33 +16,6 @@ namespace Microsoft.DotNet.ProjectJsonMigration.Tests
 {
     public class GivenThatIWantToMigrateProjectDependencies : TestBase
     {
-        // Workaround For P2P dependencies
-        // ISSUE: https://github.com/dotnet/sdk/issues/73
-        [Fact]
-        public void If_a_project_dependency_is_present_DesignTimeAutoUnify_and_AutoUnify_are_present()
-        {
-            var solutionDirectory =
-                TestAssetsManager.CreateTestInstance("TestAppWithLibrary", callingMethod: "p").Path;
-
-            var appDirectory = Path.Combine(solutionDirectory, "TestApp");
-            var libDirectory = Path.Combine(solutionDirectory, "TestLibrary");
-
-            var projectContext = ProjectContext.Create(appDirectory, FrameworkConstants.CommonFrameworks.NetCoreApp10);
-            var mockProj = ProjectRootElement.Create();
-            var testSettings = new MigrationSettings(appDirectory, appDirectory, "1.0.0", mockProj);
-            var testInputs = new MigrationRuleInputs(new[] {projectContext}, mockProj, mockProj.AddItemGroup(),
-                mockProj.AddPropertyGroup());
-            new MigrateProjectDependenciesRule().Apply(testSettings, testInputs);
-
-            var autoUnify = mockProj.Properties.Where(p => p.Name == "AutoUnify");
-            autoUnify.Count().Should().Be(1);
-            autoUnify.First().Value.Should().Be("true");
-
-            var designTimeAutoUnify = mockProj.Properties.Where(p => p.Name == "DesignTimeAutoUnify");
-            designTimeAutoUnify.Count().Should().Be(1);
-            designTimeAutoUnify.First().Value.Should().Be("true");
-        }
-
         [Fact]
         public void Project_dependencies_are_migrated_to_ProjectReference()
         {

